src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2835,6 +2835,83 @@ public override IOperation VisitCoalesce(ICoalesceOperation operation, int? capt
             return result;
         }
 
+        public override IOperation VisitCoalesceAssignment(ICoalesceAssignmentOperation operation, int? captureIdForResult)
+        {
+            SpillEvalStack();
+
+            // If we're in a statement context, we elide the capture of the result of the assignment, as it will
+            // just be wrapped in an expression statement that isn't used anywhere and isn't observed by anything.
+            bool isStatement = _currentStatement == operation || operation?.Parent.Kind == OperationKind.ExpressionStatement;
+            Debug.Assert(captureIdForResult == null || !isStatement);
+
+            RegionBuilder resultCaptureRegion = _currentRegion;
+
+            EvalStackFrame frame = PushStackFrame();
+
+            PushOperand(Visit(operation.Target));
+            SpillEvalStack();
+            IOperation locationCapture = PopOperand();
+
+            // Capture the value, as it will only be evaluated once. The location will be used separately later for
+            // the null case
+            EvalStackFrame valueFrame = PushStackFrame();
+            SpillEvalStack();
+            int valueCaptureId = GetNextCaptureId(valueFrame.RegionBuilderOpt);
+            AddStatement(new FlowCapture(valueCaptureId, locationCapture.Syntax, locationCapture));
+            IOperation valueCapture = GetCaptureReference(valueCaptureId, OperationCloner.CloneOperation(locationCapture));
```

Comment:
>IOperation valueCapture [](start = 12, length = 23)

Would it make sense to introduce ```valueCapture``` only for non-statement case? Only then we are using the value twice.

Reply:
I thought about that, but we still need a capture for the location for the when null case, and then we're back in the same problem where locationCapture is being used as both a value and a location in different sections.

---
In reply to: [212138723](https://github.com/dotnet/roslyn/pull/29063#discussion_r212138723) [](ancestors = 212138723)